### PR TITLE
Validation Errors for Forms

### PR DESCRIPTION
**Pull request by nummi**
Trying to consistificationize how we handle form errors.

Also includes some changes to Flash Messages.

### CODE REVIEW

File: app/assets/javascripts/controllers/overlays/user_detail_overlay_controller.js.coffee [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
-
     saveUser: ->
-      @get('model').save().then =>
-        @send('closeOverlay')
+      @get('model').save()
+                   .then =>
+                     @clearValidationErrors()
+                     @send('closeOverlay')
+                   .catch (response) =>
+                     @displayValidationErrorsFromResponse response
```
Comment (nummi) [inline app/assets/javascripts/controllers/overlays/user_detail_overlay_controller.js.coffee]: You have no taste, Hound.

File: app/assets/javascripts/controllers/task_controller.js.coffee [hunk 2]
```diff
@@ -27,9 +27,9 @@ ETahi.TaskController = Ember.ObjectController.extend ETahi.SavesDelayed, ETahi.C
     @_super()
       .then () =>
         @clearValidationErrors()
-      .catch (error) =>
-        @setValidationErrors(error.errors)
-        @set('model.completed', false)
+      .catch (response) =>
+        @displayValidationErrorsFromResponse response
```
Comment (nummi) [inline app/assets/javascripts/controllers/task_controller.js.coffee]: You have no taste, Hound.

File: app/assets/javascripts/controllers/journal_thumbnail_controller.js.coffee [hunk 3]
```diff
@@ -38,8 +29,12 @@ ETahi.JournalThumbnailController = Ember.ObjectController.extend ETahi.FileUploa
     saveJournalDetails: ->
       updateLogo = @get('uploadLogoFunction')
       if @get('model.isNew')
-        @get('model').save().then (journal) =>
-          (updateLogo || @stopEditing).call(@)
+        @get('model').save()
+                     .then (journal) =>
+                       (updateLogo || @stopEditing).call(@)
+                     .catch (response) =>
+                       @displayValidationErrorsFromResponse response
```
Comment (nummi) [inline app/assets/javascripts/controllers/journal_thumbnail_controller.js.coffee]: You have no taste, Hound.

File: app/assets/javascripts/controllers/task_controller.js.coffee [hunk 4]
```diff
@@ -1,4 +1,4 @@
-ETahi.TaskController = Ember.ObjectController.extend ETahi.SavesDelayed, ETahi.ControllerParticipants, ETahi.ValidatesAssociatedModels, Ember.Evented,
+ETahi.TaskController = Ember.ObjectController.extend ETahi.SavesDelayed, ETahi.ControllerParticipants, ETahi.ValidationErrorsMixin, Ember.Evented,
```
Comment (nummi) [inline app/assets/javascripts/controllers/task_controller.js.coffee]: Don't care, Hound.

**Comment by mikem** [comment]
After a skim through the changes, it looks alright to me. :+1:

File: app/assets/javascripts/controllers/journal_thumbnail_controller.js.coffee [hunk 5]
```diff
@@ -1,31 +1,22 @@
...
     @get('model').save()
-                 .then(@stopEditing.bind(@))
-                 .catch ({errors: {name, description}}) ->
-                   self.setProperties
-                     nameErrors: name?[0]
-                     descriptionErrors: description?[0]
+                 .then =>
+                   @stopEditing()
+                 .catch (response) =>
+                   @displayValidationErrorsFromResponse response
```
Comment (nummi) [inline app/assets/javascripts/controllers/journal_thumbnail_controller.js.coffee]: Whatev's, Hound.

**Pull request merged**